lifecycle-operator/controllers/lifecycle/keptntask/job_utils_test.go

```diff
@@ -201,6 +202,76 @@ func TestKeptnTaskReconciler_updateTaskStatus(t *testing.T) {
 	require.Equal(t, apicommon.StateSucceeded, task.Status.Status)
 }
 
+func TestKeptnTaskReconciler_generateJob(t *testing.T) {
+	namespace := "default"
+	taskName := "my-task"
+	svcAccname := "svcAccname"
+	taskDefinitionName := "my-task-definition"
+	token := true
+
+	taskDefinition := makeTaskDefinitionWithServiceAccount(taskDefinitionName, namespace, svcAccname, &token)
+	taskDefinition.Spec.ServiceAccount.Name = svcAccname
+	taskDefinition.Spec.Container = containerBuilder()
+
+	serviceAccount := makeServiceAccount(svcAccname, namespace, &token)
```

Comment:
I think we do not need to create a service account, we just need to fake the usage of it

Reply:
without it I couldn't create fake for it

Reply:
you do not use a service account in the `generateJob` function, just it's name and token, and these two fields you have from `keptntaskdefinition`, therefore you do not need it